Pm 31 filter pets page

asching7108 opened the pull request:
Added filters on pets page
  - page is responsive
  - fixed the pet card display issue (when there's only one pet in a row it takes all the space)
  - user preferences not handled yet

asching7108 committed:
PM-31 added filters

asching7108 committed:
PM-31 updated pet card

asching7108 commented on package.json:
```diff
@@ -22,6 +22,7 @@
     "react-password-checklist": "^1.3.3",
     "react-router-dom": "^6.2.1",
     "react-scripts": "5.0.0",
+    "react-select": "^5.2.2",
```
Added this package for multi-select

dotte-osu replied:
thanks!

asching7108 commented on src/components/PetCard/PetCard.js:
```diff
@@ -1,10 +1,12 @@
-export default function PetCard(props, newClass) {
-  const { pet } = props;
+export default function PetCard(props) {
+  const { pet, page, className } = props;
```
Added a 'page' prop since the PetCard can be used for the pets page, shelter page and favorites page but with minor differences, we can use this prop to manipulate elements

dotte-osu replied:
Good idea! Thank you!

asching7108 commented on src/components/Utils/Utils.js:
```diff
@@ -58,11 +60,20 @@ export function SecondaryButton({ className, ...props }) {
   );
 }
 
+export function renderFavoriteIcon(petID) {
```
this method is moved here because both the pet page and the pet card use it

asching7108 committed:
PM-31 removed unused imports

dotte-osu commented on src/routes/PetsPage/PetsPage.js:
```diff
@@ -1,34 +1,200 @@
 
   componentDidMount() {
-    PetsService.getPets()
+    PetsService.getBreeds()
```
Do you think we can call `getBreeds()` in the `/shelters/shelterID/pets/create` page so that a user can choose from the existing breeds?

asching7108 replied:
Definitely, I actually forgot to change it

dotte-osu commented on src/routes/PetsPage/PetsPage.js:
```diff
@@ -1,34 +1,200 @@
+import { Input, Section } from '../../components/Utils/Utils';
 import PetsService from '../../services/petsService';
 
 export default class PetsPage extends Component {
```
Is there any way to call this `PetsPage` from `http://localhost:3000/shelters/shelterID` page? The shelter page also has pet list and if would be nice if we can just call `PetsPage`  from the shelter page with the shelter pets data. I really like the way you set the filters and it would be nice if we can use it in the below page!
![image](https://user-images.githubusercontent.com/71732016/153928098-f03bc872-33a6-4898-9da3-829a340c659e.png)

asching7108 replied:
Yes I should be able to change it to a component!

dotte-osu replied:
Thank you! I am excited to see your change!!

dotte-osu reviewed (changes requested):
Great work!! I love the way you implemented the filters! The `/pets` page looks great! I let a couple of question regarding the typeofbreeds and `PetsPage`. Let me know if you have any questions.

asching7108 committed:
PM-31 changed add pet page breed input to select

asching7108 committed:
PM-31 refactored pet list and pet filters

asching7108 committed:
PM-31 reused pet list for shelter page

asching7108 committed:
PM-31 removed unused file

asching7108 commented:
PR is ready for review!

I created a PetList component that is used in PetsPage and ShelterPage. Also removed some folders under /components that have only one file (and probably won't have a .css file).
Also modified the backend so that the GET /pets endpoint now also takes a query for shelterID to filters the pets for the shelter page.

asching7108 commented on src/routes/PetsPage/PetsPage.js:
```diff
@@ -1,54 +1,90 @@
+      maxAge,
+      more
+    } = this.state;
+    const filters = { availability: 'Available' };
```
I think for users they only need to see pets that are currently available. For shelter admins I'm planning to add an Availability filter. I'll address that when I get to work on the edit pets feature. Let me know if you think otherwise!

dotte-osu commented:
Thank you! Both /pets page and /shelter/id/pets page looks great!! Please fix the merge conflict and you are good to go!

asching7108 committed:
Merge branch 'main' into PM-31-filter-pets-page

asching7108 commented:
conflicts resolved

dotte-osu commented:
Thanks! Approved!
